### PR TITLE
Users management

### PR DESCRIPTION
**Pull request by GillesRasigade**
Script to configure and manage users on a MongoDb database.

This is script allows to:
- create the roles on the database: CP_READ, CP_WRITE
- update roles after an update
- add and update a user with specific roles
- revoke a user

### CODE REVIEW

File: mongodb_manage_users.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# logger.info('hello')
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Dumps heroku variables into an environment description')
+    parser.add_argument('target', type=str, help='name of the environment')
+    parser.add_argument('action', type=str, help='action to perform')
+
+    parser.add_argument('-d', '--db', type=str, help='database name')
+    parser.add_argument('-u', '--name', type=str, help='name for a USER or for a ROLE')
+    parser.add_argument('-p', '--password', type=str, help='paddword')
```
Comment (rossille) [inline mongodb_manage_users.py]: password

File: lib/mongodb.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+from pymongo import MongoClient, errors
+
+from . import config, sys_util
+
+
+
+class MongodbGateway():
+    CONST_ROLES = [
+        {
+        'role': 'CP_READ',
```
Comment (rossille) [inline lib/mongodb.py]: hard to read: indentation += 4 please :=)

File: lib/mongodb.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        'roles': []
+        },
+        {
+        'role': 'CP_WRITE',
+            'privileges': [{
+                'resource': { 'db': '', 'collection': '' },
+                'actions': [
+                    'insert',
+                    'update',
+                    # 'remove' # NOT NECESSARY FOR PRODUCTION PURPOSE
```
Comment (rossille) [inline lib/mongodb.py]: I know that a few features of some projects require the "remove" action. 
Although it's for bad reasons, I wouldn't want to see all the require fixes to stop depending on destructive actions on the critical path of removing the worst actions like "dropCollections" and "dropDatabase", so I would like to see a "CP_LEGACY_WRITE" too.

File: lib/mongodb.py (hunk 2 above)
Comment (rossille) [inline lib/mongodb.py]: also, about the name CP_READ, I would like to remove the 'CP_' prefix: letsdeploy is opensource and aims at being generic. So let's remove the CP_ prefix and make sure we configure the role as what we thing are the best pratice for a regular readonly role and a regular readwrite role for a standard application

**Comment by GillesRasigade** [comment]
Concerning the tests, I will change the permissions

**Pull request merged**